### PR TITLE
Fix #18323 - honor anal.calls in aa ##anal

### LINKED ISSUE #18323: aa ignores anal.calls
tried today on master

### PR DESCRIPTION
**Pull request by trufae**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

### CODE REVIEW

**Comment by trufae** [comment]
Now its using the rbin info to set the function names instead of the flags. Flags may have the realname already but rbin doesnt know the flagname.. opinions here?

**Comment by trufae** [comment]
C72A6D8F-EC13-4649-AE2D-E7ED25EFC771

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -164,12 +164,18 @@ static bool isInvalidMemory(RAnal *anal, const ut8 *buf, int len) {
...
 }
 
-static bool isSymbolNextInstruction(RAnal *anal, RAnalOp *op) {
-	r_return_val_if_fail (anal && op && anal->flb.get_at, false);
+static bool is_symbol_flag(const char *name) {
+	return strstr (name, "imp.")
+		|| strstr (name, "dbg.")
+		|| strstr (name, "sym.")
+		|| strstr (name, "entry")
+		|| strstr (name, "main");
```
Comment (lowlyw) [inline libr/anal/fcn.c:172]: shouldn't this be `strncmp` because `main` and `entry` should be the beginning?

Comment (lowlyw) [reply libr/anal/fcn.c:172]: also don't we have the symbols listed, this seems kind of janky?

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -870,36 +870,33 @@ static void __add_vars_sdb(RCore *core, RAnalFunction *fcn) {
...
-			!r_strbuf_setf (&value, "%d", arg_count)) {
-			goto exit;
-		}
-		sdb_set (core->anal->sdb_types, r_strbuf_get (&key), r_strbuf_get (&value), 0);
-	}
-exit:
-	r_strbuf_fini (&key);
-	r_strbuf_fini (&value);
+			char *name = var->name;
+			char *k = r_str_newf ("func.%s.arg.%zd", fcn->name, arg_count++);
```
Comment (lowlyw) [inline libr/core/cmd_anal.c:882]: `%zu` instead?

File: libr/core/cmd_anal.c [hunk 3]
```diff
@@ -4211,122 +4307,28 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
...
-		}
-		if (name) {
-			if (*name && !__setFunctionName (core, addr, name, true)) {
-				eprintf ("af: Cannot find function at 0x%08" PFMT64x "\n", addr);
+			bool anal_calls = false;
+			if (input[0] && input[1] == 'r') {
+				input++;
+				anal_calls = true;
+			} else {
+				anal_calls = r_config_get_i (core->config, "anal.calls");;
```
Comment (lowlyw) [inline libr/core/cmd_anal.c:4315]: extra semicolon

**Review by lowlyw — CHANGES_REQUESTED** [review]

File: libr/anal/fcn.c (hunk 1 above)
Comment (trufae) [reply libr/anal/fcn.c:172]: yeah correct, that may solve some false positives i guess.

**Comment by trufae** [comment]
As long as this change may affect analysis and introduce new issues im moving it for 5.3, 5.2 is an important stability release and shouldnt take longer to happen

**Comment by trufae** [comment]
The only test taht is failing now is : 
Screenshot 2021-04-24 at 23 31 23

**Pull request merged**